fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -0,0 +1,78 @@
+module FastlaneCore
+  class BuildWatcher
+    def self.find_build(app = nil)
+      build = nil
+      app.latest_version.candidate_builds.each do |b|
+        if !build or b.upload_date > build.upload_date
+          build = b
+        end
+      end
+      unless build
+        UI.user_error!("No processing builds available for app #{app.bundle_id}")
+      end
+      return build
+    end
+
+    def self.wait_for_train(app = nil, platform = "ios", sleep_time = 30)
```

Comment:
i am unsure about that.
how'd i make a instance method private?

`private_class_method :wait_for_train` - will that work on older ruby's ?